Directory.cpp

```diff
@@ -43,36 +47,60 @@ Mappa* Mappa::search(string dirName) {
 	return nullptr;
 }
 
-bool Mappa::makeDirectory(string dirName) {
+bool Directory::makeDirectory(string dirName) {
 	if (this->search(dirName) == nullptr) {
-		content.push_back(new Mappa(dirName));
+		content.push_back(new Directory(dirName));
 		return true;
 	}
 	return false;
 }
 
-bool Mappa::hasChild() {
-	return content.size() > 0;
+bool Directory::hasChild() {
+	return content.size() > 0 || files.size() > 0;
 }
 
-bool Mappa::removeDirectory(string dirName) {
-	Mappa* iter = this->search(dirName);
+bool Directory::remove(string name) {
+	File* f = this->searchFile(name);
+	if (f != nullptr) {
+		files.remove(f);
+		return true;
+	}
+
+	Directory* iter = this->search(name);
 	bool Child = iter->hasChild();
 
 	if (iter != nullptr && Child == false) {
 		content.remove(iter);
 		return true;
 	}
 	else { return false; }
-
-
 }
 
-bool Mappa::removeRecursiveDirectory(string dirName) {
-	Mappa* iter = this->search(dirName);
+bool Directory::removeRecursiveDirectory(string dirName) {
+	files.clear();
+	Directory* iter = this->search(dirName);
 	iter->recurciveDelete();
 	content.remove(iter);
 
 	return true;
 

```

Comment:
Na ezt most nem teljesen ertem. Ha ez arar van, hogy a `dirName` alkonyvtarat rekurzivan kitorolje, akkro miert kell mellole az elso sorban kitorolni a fajlokat is? Ha meg nem arra van, hanem rekurzivan mindent ki kell torolni, akkor hol a for ciklus? Ez most en azt latom, hogy ha megadom egyszer, hogy `a` akkor addig kitorol minden fajlt, amig tud `a` alkonyvtarba bemenni. 
+1: mi a megallasi feltetel? mi van, ha az iter az nullptr?

Reply:
Átdolgoztam a teljes törlési gondolatmenetet és az ahhoz kapcsolódó tagfgv-eket. Így átláthatóbb és jól is működik már.